File: ALODWebUtility/Common/ChangeRow.cs
using System;

namespace ALODWebUtility.Common
{
    [Serializable]
    public class ChangeRow
    {
        protected DateTime _actionDate;
        protected int _actionId;
        protected string _actionName;
        protected string _field;
        protected int _id;
        protected string _newVal;
        protected string _oldVal;
        protected string _section;
        protected int _userId;
        protected string _userName;

        public ChangeRow()
        {
            _id = 0;
            _section = string.Empty;
            _field = string.Empty;
            _oldVal = string.Empty;
            _newVal = string.Empty;
        }

        public ChangeRow(int id, string section, string field, string oldVal, string newVal)
        {
            _id = id;
            _section = section;
            _field = field;
            _oldVal = oldVal;
            _newVal = newVal;
        }

        public ChangeRow(string section, string field, string oldVal, string newVal)
        {
            _id = Id; // This looks like a bug in the original VB code, assigning property Id to field _id, but Id uses _id. 
                      // In VB, Id property access inside the class might refer to the property. 
                      // But _id is initialized to 0 in default constructor. 
                      // In the VB code: _id = Id. Id returns _id. So _id = _id.
                      // I will keep it as is, but it's redundant.
            _section = section;
            _field = field;
            _oldVal = oldVal;
            _newVal = newVal;
        }

        public DateTime ActionDate
        {
            get { return _actionDate; }
            set { _actionDate = value; }
        }

        public int ActionId
        {
            get { return _actionId; }
            set { _actionId = value; }
        }

        public string ActionName
        {
            get { return _actionName; }
            set { _actionName = value; }
        }

        public string Field
        {
            get { return _field; }
            set { _field = value; }
        }

        public int Id
        {
            get { return _id; }
            set { _id = value; }
        }

        public string NewVal
        {
            get { return _newVal; }
            set { _newVal = value; }
        }

        public string OldVal
        {
            get { return _oldVal; }
            set { _oldVal = value; }
        }

        public string Section
        {
            get { return _section; }
            set { _section = value; }
        }

        public int UserId
        {
            get { return _userId; }
            set { _userId = value; }
        }

        public string UserName
        {
            get { return _userName; }
            set { _userName = value; }
        }
    }
}

File: ALODWebUtility/Common/ChangeRow.vb
Namespace Common

    <Serializable()>
    Public Class ChangeRow

        Protected _actionDate As DateTime
        Protected _actionId As Integer
        Protected _actionName As String
        Protected _field As String
        Protected _id As Integer
        Protected _newVal As String
        Protected _oldVal As String
        Protected _section As String
        Protected _userId As Integer
        Protected _userName As String

        Public Sub New()
            _id = 0
            _section = String.Empty
            _field = String.Empty
            _oldVal = String.Empty
            _newVal = String.Empty
        End Sub

        Public Sub New(ByVal id As Integer, ByVal section As String, ByVal field As String, ByVal oldVal As String, ByVal newVal As String)
            _id = id
            _section = section
            _field = field
            _oldVal = oldVal
            _newVal = newVal
        End Sub

        Public Sub New(ByVal section As String, ByVal field As String, ByVal oldVal As String, ByVal newVal As String)
            _id = Id
            _section = section
            _field = field
            _oldVal = oldVal
            _newVal = newVal
        End Sub

        Public Property ActionDate() As DateTime
            Get
                Return _actionDate
            End Get
            Set(ByVal value As DateTime)
                _actionDate = value
            End Set
        End Property

        Public Property ActionId() As Integer
            Get
                Return _actionId
            End Get
            Set(ByVal value As Integer)
                _actionId = value
            End Set
        End Property

        Public Property ActionName() As String
            Get
                Return _actionName
            End Get
            Set(ByVal value As String)
                _actionName = value
            End Set
        End Property

        Public Property Field() As String
            Get
                Return _field
            End Get
            Set(ByVal value As String)
                _field = value
            End Set
        End Property

        Public Property Id() As Integer
            Get
                Return _id
            End Get
            Set(ByVal value As Integer)
                _id = value
            End Set
        End Property

        Public Property NewVal() As String
            Get
                Return _newVal
            End Get
            Set(ByVal value As String)
                _newVal = value
            End Set
        End Property

        Public Property OldVal() As String
            Get
                Return _oldVal
            End Get
            Set(ByVal value As String)
                _oldVal = value
            End Set
        End Property

        Public Property Section() As String
            Get
                Return _section
            End Get
            Set(ByVal value As String)
                _section = value
            End Set
        End Property

        Public Property UserId() As Integer
            Get
                Return _userId
            End Get
            Set(ByVal value As Integer)
                _userId = value
            End Set
        End Property

        Public Property UserName() As String
            Get
                Return _userName
            End Get
            Set(ByVal value As String)
                _userName = value
            End Set
        End Property

    End Class

End Namespace